lib/logstash/plugin_mixins/http_client.rb

```diff
@@ -94,6 +94,13 @@ def setup_http_client_config
     # 2. Proxy host in form: `{host => "proxy.org", port => 80, scheme => 'http', user => 'username@host', password => 'password'}`
     # 3. Proxy host in form: `{url =>  'http://proxy.org:1234', user => 'username@host', password => 'password'}`
     config :proxy
+
+    # If you'd like to use authentication. Options available include:
+    #
+    # user     - username to be used
+    # password - password to be used
+    # eager    - eagerly offer the Authorization header before the server challenges for it
+    config :auth
```

Comment:
IMHO I would keep the configuration in separated fields like

     # Basic Auth - username
     config :user, :validate => :string

     # Basic Auth - password
     config :password, :validate => :password

This would align with elasticsearch input/filter/output plugins configuration style and avoid for the password to be shown in logstash debug logs
Also I would opt out the eager parameter for now, unless you can explain me concrete use case for it, because from Manticore doc, it is not a standard option. If kept, then keeping the warning in the doc feels necessary `You should only use this if you have a specific need for it, as it may be a security concern otherwise.`